Yoast/wordpress-seo
# Sort the SEO and readability analysis results alphabetically

## Summary

This PR can be summarized in the following changelog entry:

* Improves the consistency of the SEO and readability results by showing them in a fixed order. 

## Relevant technical choices:

* Sorts the results by the `_identifier ` that each of the assessments has.

## Test instructions
This PR can be tested by following these steps:

* The results are not sorted alphabetically by the line of text they display in the results, they are sorted by their `_identifier `. This means for testing it would be easiest to check the Redux Store to see the identifier. 
* Build all the files. 
* Create or open a post with some text, so that a decent SEO and Readability Analysis is shown. 
* install Redux dev tools if you haven't yet: `https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en`
* Inspect the page, go to `Redux`, click "Core/Data" and switch to "yoast-seo/editor". In the right top, click "State" and in the data below click "Analysis". Here you will see "Seo" and "Readability". In here open "results" and you see they are sorted alphabetically by their identifier. 

For testing without Redux (which is unavailable without Gutenberg installed) you can use the data below. It shows analysis results combined with their _identifier. A result with identifier "externalLinks" should always show somewhere on top, as it is first in the alphabet. Mind that it can be in one of the 3 different results (red, green, orange). A sentence with identifier "urlKeyword" should always be shown last as it it at the end of the alphabet. 

SEO analysis: 
```
 text: "This page has 0 nofollowed <a href='https://yoa.st/2pl' target='_blank'>outbound link(s)</a> and 2 normal outbound link(s)." _identifier: "externalLinks"  
text: "This page has 1 <a href='https://yoa.st/2pm' target='_blank'>internal link(s)</a>." _identifier: "internalLinks"  
text: "All topic words appear within one sentence in the <a href='https://yoa.st/2pc' target='_blank'>first paragraph</a> of the copy." _identifier: "introductionKeyword" 
 text: "Your <a href='https://yoa.st/2pdd' target='_blank'>keyphrase</a> has a nice length." _identifier: "keyphraseLength"  
text: "The exact-match <a href='https://yoa.st/2pe' target='_blank'>keyword density</a> is 1.5%, which is great; the focus keyword was found 6 times." _identifier: "keywordDensity"  
text: "The focus keyword 'Dutch translation services' does not appear in the <a href='https://yoa.st/2pn' target='_blank'>SEO title</a>." _identifier: "titleKeyword"
  text: "The <a href='https://yoa.st/2po' target='_blank'>SEO title</a> is too short. Use the space to add keyword variations or create compelling call-to-action copy." _identifier: "titleWidth" 
 text: "The focus keyword does not appear in the <a href='https://yoa.st/2pp' target='_blank'>URL</a> for this page. If you decide to rename the URL be sure to check the old URL 301 redirects to the new one!" _identifier: "urlKeyword" 
```
Readability Analysis: 
```
text: "The copy scores 67.8 in the <a href='https://yoa.st/flesch-reading' target='_blank'>Flesch Reading Ease</a> test, which is considered ok to read. " _identifier: "fleschReadingEase"  
text: "8.6% of the sentences contain <a href='https://yoa.st/passive-voice' target='_blank'>passive voice</a>, which is less than or equal to the recommended maximum of 10%." _identifier: "passiveVoice"
  text: "Great job with using <a href='https://yoa.st/headings' target='_blank'>subheadings</a>!" _identifier: "subheadingsTooLong"  
text: "None of the paragraphs are too long, which is great." _identifier: "textParagraphTooLong"  
text: "5.7% of the sentences contain <a href='https://yoa.st/short-sentences' target='_blank'>more than 20 words</a>, which is less than or equal to the recommended maximum of 25%." _identifier: "textSentenceLength"  
text: "48.6% of the sentences contain a <a href='https://yoa.st/transition-words' target='_blank'>transition word</a> or phrase,  _identifier: "textTransitionWords" 
```
## Quality assurance


Fixes #11033
